cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -470,17 +529,17 @@ angular.module(PKG.name + '.commons')
 
       let sourceIdSplit;
 
-      if (sourceIsCondition || sourceIsPort) {
+      if (sourceIsPort) {
+        let portClass = getPortEndpointClass(newConnObj.source.classList);
+        sourceIdSplit = portClass.split('_');
+        connection.port = sourceIdSplit[2];
+      } else {
```

Comment:
This could be 

```
else if (sourceIsCondition) {
 ... 
}
```
could be easier to read.

Reply:
Actually this condition does need to contain all cases of `if (!sourceIsPort)`. I do have a check specifically for `sourceIsCondition` inside this `else`.

Reply:
Hi there, 
these are colors we can use from our palette:
@spark-plugins-color: #454A58
 @error-transform: #D40001;
-------------------------------
+@alertpublisher-plugins-color: #ffba01; (this color is already part of the palette)
@sink-plugins-color: #8367df; (we don't have a bright purple in our palette, I suggest to keep as is)